### PR TITLE
Issue #5927: PHP 8.2 compatibility

### LINKED ISSUE #5927: Make Backdrop core compatible with PHP 8.2
Our Drupal brethren have started some work on this in a meta: https://www.drupal.org/project/drupal/issues/3304807

### PR DESCRIPTION
**Pull request by klonos**
Fixes https://github.com/backdrop/backdrop-issues/issues/5927

Lets see how many things break 😬

### CODE REVIEW

File: core/modules/entity/entity.query.inc [hunk 1]
```diff
@@ -947,7 +947,7 @@ class EntityFieldQuery {
       return $select_query->countQuery()->execute()->fetchField();
     }
     $return = array();
-    $this->orderedResults = array();
+    $this->ordered_results = array();
```
Comment (indigoxela) [inline core/modules/entity/entity.query.inc:950]: Out of curiosity: why switching from camelCase to underscore?

Shouldn't we use camelCase for properties and methods, wherever possible, and underscores only, when inevitable for BC reasons (like classes that get extended in contrib...)?

File: core/modules/comment/views/views_handler_argument_comment_user_uid.inc [hunk 2]
```diff
@@ -10,6 +10,7 @@
  *
  * @ingroup views_argument_handlers
  */
+#[\AllowDynamicProperties]
 class views_handler_argument_comment_user_uid extends views_handler_argument {
```
Comment (indigoxela) [inline core/modules/comment/views/views_handler_argument_comment_user_uid.inc:14]: Wouldn't it make more sense to set `#[\AllowDynamicProperties]` only once for the base class views_handler_argument? If I got things right, this is inherited.

Edit: maybe class views_object is the one to fix most problems with the least effort and smallest change.

File: core/modules/comment/views/views_handler_filter_comment_user_uid.inc [hunk 3]
```diff
@@ -10,6 +10,7 @@
  *
  * @ingroup views_filter_handlers
  */
+#[\AllowDynamicProperties]
 class views_handler_filter_comment_user_uid extends views_handler_filter_user_name {
```
Comment (indigoxela) [inline core/modules/comment/views/views_handler_filter_comment_user_uid.inc:14]: I think, the base class for that class is views_handler_filter (modules/views/handlers/views_handler_filter.inc).

File: core/modules/entity/entity.query.inc (hunk 1 above)
Comment (herbdool) [reply core/modules/entity/entity.query.inc:950]: Agreed. Plus we can't change a public method without a very good reason to break compatibility.

File: core/includes/database/select.inc [hunk 4]
```diff
@@ -972,14 +972,14 @@ class SelectQuery extends Query implements SelectQueryInterface {
    *
    * @var array
    */
-  protected $alterTags = array();
+  public $alterTags = array();
```
Comment (indigoxela) [inline core/includes/database/select.inc:975]: Switching visibility doesn't seem right to me. Why that?

Comment (herbdool) [reply core/includes/database/select.inc:975]: Can't change the visibility of properties

**Comment by klonos** [comment]
Closing this in favor of https://github.com/backdrop/backdrop/pull/4445

**Pull request closed without merging**